internal/agent/device/console/console.go

```diff
@@ -204,31 +212,61 @@ func (c *ConsoleController) startForwarding(ctx context.Context, stdin io.WriteC
 }
 
 func (c *ConsoleController) bashProcess(ctx context.Context) (io.WriteCloser, io.ReadCloser, error) {
-	// TODO pty: this is how oci does a PTY:
-	// https://github.com/cri-o/cri-o/blob/main/internal/oci/oci_unix.go
-	//
-	// set PS1 environment variable to make bash print the default prompt
+
+	// create a new bash process
 	cmd := c.executor.CommandContext(ctx, "bash", "-i", "-l")
-	stdin, err := cmd.StdinPipe()
-	if err != nil {
-		return nil, nil, fmt.Errorf("error getting stdin pipe: %w", err)
-	}
-	stdout, err := cmd.StdoutPipe()
+	cmd.Env = append(cmd.Env, "TERM=xterm-256color")
+
+	// create a new PTTY
+	p, err := pty.Start(cmd)
 	if err != nil {
-		return nil, nil, fmt.Errorf("error getting stdout pipe: %w", err)
+		return nil, nil, fmt.Errorf("error starting ptty: %w", err)
 	}
 
-	cmd.Stderr = cmd.Stdout
+	c.log.Info("bash process started under ptty")
+
+	stdinReader, stdinWriter := io.Pipe()
+	stdoutReader, stdoutWriter := io.Pipe()
+
+	var stdinErr, stdoutErr error
 
-	if err := cmd.Start(); err != nil {
-		return nil, nil, fmt.Errorf("error starting bash process: %w", err)
+	go func() {
+		// copy from stdinWriter to ptty
+		_, stdinErr = pools.Copy(p, stdinReader)
+		_ = cmd.Process.Kill()
+	}()
+	go func() {
+		// copy from ptty to stdoutWriter
+		_, stdoutErr = pools.Copy(stdoutWriter, p)
+		stdoutWriter.Close()
+		_ = cmd.Process.Kill()
+	}()
+
+	// set the initial terminal size until we receive a window resize event from the other side
+	size := TerminalSize{Width: 80, Height: 25}
+	if err := setSize(p.Fd(), size); err != nil {
+		return nil, nil, fmt.Errorf("error setting terminal size: %w", err)
 	}
+
 	go func() {
-		if err := cmd.Wait(); err != nil {
+		defer p.Close()
+		err := cmd.Wait()
+		if stdinErr != nil {
+			c.log.Errorf("error copying to stdin: %v", stdinErr)
+		}
+		if stdoutErr != nil {
+			c.log.Errorf("error copying from stdout: %v", stdoutErr)
+		}
+		if err != nil {
 			c.log.Errorf("error waiting for bash process: %v", err)
 		} else {
 			c.log.Info("bash process exited successfully")
 		}
 	}()
-	return stdin, stdout, nil
+	return stdinWriter, stdoutReader, nil
+}
+
+func setSize(fd uintptr, size TerminalSize) error {
```

Comment:
Are resize events going to be handled in this PR or some follow-up ?

Reply:
Follow up , I want to make sure we have it ASAP in, including the spec/selinux changes so the rpm side of things can incorporate that downstream.